scripts/ci/lz4-block-codec.wat

```diff
@@ -0,0 +1,384 @@
+                i32.const 255
+                i32.eq
+                br_if 0
+            end end
+            get_local $outPtr           ;; copy literals to output buffer
+            get_local $inPtr
+            get_local $clen
+            call $copy
+            get_local $outPtr           ;; advance output buffer pointer past copy
+            get_local $clen
+            i32.add
+            set_local $outPtr
+            get_local $clen             ;; advance input buffer pointer past literals
+            get_local $inPtr
+            i32.add
+            tee_local $inPtr
+            get_local $inPtrEnd         ;; exit if this is the last sequence
+            i32.eq
+            br_if $noMoreSequence
+        end
+        get_local $outPtr               ;; read match offset
+        get_local $inPtr
+        i32.load8_u
+        get_local $inPtr
+        i32.load8_u offset=1
+        i32.const 8
+        i32.shl
+        i32.or
+        i32.sub
+        tee_local $matchPtr
+        get_local $outPtr               ;; match position can't be outside input buffer bounds
+        i32.eq
+        br_if $noMoreSequence
+        get_local $matchPtr
+        get_local $inPtrEnd
+        i32.lt_u
+        br_if $noMoreSequence
+        get_local $inPtr                ;; advance input pointer past match offset bytes
+        i32.const 2
+        i32.add
+        set_local $inPtr
+        get_local $token                ;; extract length of match from token
+        i32.const 15
+        i32.and
+        i32.const 4
+        i32.add
+        tee_local $clen
+        i32.const 19                    ;; consume extra length bytes if present
+        i32.eq
+        if loop
+            get_local $inPtr
+            i32.load8_u
+            get_local $inPtr
+            i32.const 1
+            i32.add
+            set_local $inPtr
+            tee_local $_
+            get_local $clen
+            i32.add
+            set_local $clen
+            get_local $_
+            i32.const 255
+            i32.eq
+            br_if 0
+        end end
+        get_local $outPtr               ;; copy match to output buffer
+        get_local $matchPtr
+        get_local $clen
+        call $copy
+        get_local $clen                 ;; advance output buffer pointer past copy
+        get_local $outPtr
+        i32.add
+        set_local $outPtr
+        br 0
+    end end $noMoreSequence
+    get_local $outPtr                   ;; return number of written bytes
+    get_local $outPtr0
+    i32.sub
+)
+
+;;
+;; Copy n bytes from source to destination.
+;;
+;; It is overlap-safe only from left-to-right -- which is only what is
+;; required in the current module.
+;;
+(func $copy
+    (param $dst i32)
+    (param $src i32)
+    (param $len i32)
+    block $lessThan8 loop
+        get_local $len
+        i32.const 8
+        i32.lt_u
+        br_if $lessThan8
+        get_local $dst
+        get_local $src
+        i32.load8_u
+        i32.store8
```

Comment:
Why is this copy feature working off 8-bit copies and not 64-bit copies unless the last chunk of <=64-bit?

Reply:
It's probably left-to-right overlap safety. I think lz4 allows weird copies...